src/main/java/net/minecraftforge/fmp/capabilities/CapabilityWrapperRegistry.java

```diff
@@ -0,0 +1,34 @@
+package net.minecraftforge.fmp.capabilities;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import net.minecraftforge.common.capabilities.Capability;
+
+public class CapabilityWrapperRegistry
```

Comment:
Where exactly? The comment is on line 9 and that's just the class declaration lol.
I get what you mean, though. Wrapping all the caps when they are requested... But that's how caps work, you don't know if they have changed until you check.
What else do you want me to do? Put all the implementations into a weak set alongside the wrapped version and whenever it's requested, check if the amount of implementations and their hashcodes match, if so return the cached one, if not gen a new one and cache it? It's... doable... but it'd be messy af.